common/spec/dependabot/file_fetchers/base_spec.rb

```diff
@@ -1313,4 +1319,77 @@ def fetch_files
       end
     end
   end
+
+  context "with repo_contents_path" do
+    let(:repo_contents_path) { Dir.mktmpdir }
+    after { FileUtils.rm_rf(repo_contents_path) }
+
+    # `git clone` against a file:// URL that is filled by the test
+    let(:repo_path) { Dir.mktmpdir }
+    after { FileUtils.rm_rf(repo_path) }
+    let(:fill_repo) {}
+    before do
+      `git init #{repo_path}`
+      fill_repo
+      `git -C #{repo_path} add .`
+      `git -C #{repo_path} commit --allow-empty -m'fake clone source'`
```

Comment:
Nit: Could wrap this in `Dir.chdir` and change to `File.write("requirements.txt", contents)` in fill_repo
```suggestion
      Dir.chdir(repo_path) do
        `git init .`
        fill_repo
        `git add .`
        `git commit --allow-empty -m'fake clone source'`
      end
```

Reply:
Noice! The only wrinkle here is that since `contents` is loaded from a fixture, changing the working directory before `fill_repo` loses `./fixtures/...`.

I'm going to patch by not using a fixture for `contents`;  that was just cargo culted from the `github` fixtures above.

Reply:
Ah yes missed that, you could also `let!(:contents)` having it execute before the `it` is run